src/SignatureParameters.php

```diff
@@ -10,11 +10,12 @@ class SignatureParameters
      * @param HeaderList         $headerList
      * @param Signature          $signature
      */
-    public function __construct($key, $algorithm, $headerList, $signature)
+    public function __construct($key, $algorithm, $headerList, $signatureHeadersSpecified, $signature)
```

Comment:
This new `signatureHeadersSpecified` argument might be unnecessary. We already have the `headerList`, should we able to just check `$this->headerList->string()` (below) returns something?

Reply:
Reverted to original parameters, HeaderList is now constructed with an optional variable (defaults to original behaviour) to track if headers were explicitly specified and adapted calling functions to use this. 7aff02bd0080166bedb2b6cdd528a36018877534